### PR TITLE
Bug busting

### LINKED ISSUE #16: Hash doesn't hash values
As title.

### LINKED ISSUE #17: `profile` doesn't pass n_runs and top_n into `hotspots`
https://github.com/rayanht/paprika/blob/58f2e77d9fe52b556f33605de30bd1bd4a636741/paprika/benchmark.py#L139

### PR DESCRIPTION
**Pull request by rayanht**
Fixes #16
Fixes #17

### CODE REVIEW

File: paprika/oo.py [hunk 1]
```diff
@@ -84,7 +84,8 @@ def __eq__(self, other):
         return same_class and same_attrs
 
     def __hash__(self):
-        attributes = tuple(sorted(tuple(getattr(self, "__dict__").keys())))
+        attributes = tuple([getattr(self, "__dict__")[key] for key in
```
Comment (lhl2617) [inline paprika/oo.py:87]: seems like something `black` doesn't like -- other files have formatting issues as well. Black PR first?

**Review by lhl2617 — COMMENTED** [review]
image

One PR for each issue maybe :D

**Review by lhl2617 — APPROVED** [review]
image

One PR for each issue maybe :D

File: paprika/oo.py (hunk 1 above)
Comment (lhl2617) [reply paprika/oo.py:87]: Also IMHO prefer
```
dict_keys = getattr(self, "__dict__").keys()
attributes = tuple(...)
```

File: paprika/benchmark.py [hunk 2]
```diff
@@ -138,5 +138,5 @@ def wrapper_hotspots(*args, **kwargs):
 
 def profile(decorated_fn=None, *, n_runs=1, top_n=10):
     decorated_class = access_counter(decorated_fn)
-    decorated_class = hotspots(decorated_class)
+    decorated_class = hotspots(decorated_class, n_runs=n_runs, top_n=top_n)
```
Comment (lhl2617) [inline paprika/benchmark.py:141]: To silence linter and make sure the args actually get passed you can mock hotspots, but ideally you'd have a test that ensures that covers hotspots func as well.

Comment (rayanht) [reply paprika/benchmark.py:141]: The main issue with the test coverage of `@hotspots` is that we can't reliably/meaningfully compare the `cProfiler` output. It's probably gonna have to be a mock indeed

**Pull request merged**